src/file.cr

```diff
@@ -237,25 +240,83 @@ class File < IO::FileDescriptor
     basename(path).chomp(suffix)
   end
 
+  private alias User = System::User | String | UInt32 | Int32
+  private alias Group = System::Group | String | UInt32 | Int32
+
   # Changes the owner of the specified file.
   #
   # ```
-  # File.chown("/foo/bar/baz.cr", 1001, 100)
-  # File.chown("/foo/bar", gid: 100)
+  # File.chown("/foo/bar/baz.cr", "a_owner", "a_group")
+  # File.chown("/foo/bar", group: "a_group")
   # ```
   #
   # Unless *follow_symlinks* is set to `true`, then the owner symlink itself will
   # be changed, otherwise the owner of the symlink destination file will be
   # changed. For example, assuming symlinks as `foo -> bar -> baz`:
   #
   # ```
-  # File.chown("foo", gid: 100)                        # changes foo's gid
-  # File.chown("foo", gid: 100, follow_symlinks: true) # changes baz's gid
+  # File.chown("foo", group: "a_group")                        # changes foo's group to "a_group"
+  # File.chown("baz", group: "a_group", follow_symlinks: true) # changes baz's group to "a_group"
   # ```
-  def self.chown(path, uid : Int = -1, gid : Int = -1, follow_symlinks = false)
+  #
+  # Raises on error.
+  def self.chown(path : String, owner : User? = nil, group : Group? = nil, follow_symlinks : Bool = false) : Nil
+    Crystal::System::File.chown(path, uid_from_user(owner), gid_from_group(group), follow_symlinks)
+  end
+
+  # :nodoc:
+  def self.chown(path : String, uid : Int = -1, gid : Int = -1, follow_symlinks : Bool = false) : Nil
     Crystal::System::File.chown(path, uid, gid, follow_symlinks)
   end
 
+  # Same as `chown()` but instead returns a `Bool` indicating success.
+  # ```
+  def self.chown?(path : String, owner : User? = nil, group : Group? = nil, follow_symlinks : Bool = false) : Bool
+    Crystal::System::File.chown?(path, uid_from_user(owner), gid_from_group(group), follow_symlinks)
+  end
+
+  # :nodoc:
+  private def self.uid_from_user(user : System::User) : UInt32
+    user.uid
+  end
+
+  # :nodoc:
+  private def self.uid_from_user(user : String) : UInt32
+    System::User.uid(user)
+  end
+
+  # :nodoc:
+  private def self.uid_from_user(user : Int) : UInt32
+    raise "Invalid uid (#{user}) provided" if user < 0
+    return user.to_u32
+  end
+
+  # :nodoc:
+  private def self.uid_from_user(user : Nil) : Int32
```

Comment:
I'm not fond of returning a `UInt32 | Int32` union. Are there use cases to support a nil user/group and an invalid `-1` uid?

Reply:
In libc -1 is taken to mean don't change this value. I would prefer that crystal takes `nil` to mean that. I think it would make more intuitive sense.

More broadly these methods only exist to simplify the translation of a given argument into the expected libc argument. Is there a simpler way of going about this that is not a pair of `case`'s?